src/rdkafka_conf.c

```diff
@@ -585,28 +592,28 @@ static const struct rd_kafka_property rd_kafka_properties[] = {
           .copy = rd_kafka_conf_interceptor_copy },
 
         /* Global client group properties */
-        { _RK_GLOBAL|_RK_CGRP, "group.id", _RK_C_STR,
+        { _RK_GLOBAL|_RK_CONSUMER|_RK_CGRP, "group.id", _RK_C_STR,
```

Comment:
This is not necessarily true in the future, the client group mechanism in the Kafka protocol is not tied to the consumer but is generic.

Reply:
I'm not familiar with the internal protocol, but currently any of the settings above are for consumers only. IF (and this is a big IF) they will add some grouping to producers (seems weird conceptually), you can then have options for _RK_GLOBAL|_RK_PRODUCER|_RK_CGRP. I believe it makes the code clearer to leave these flags as-is.